interview/data_structure/binary_search_tree.py

```diff
@@ -0,0 +1,116 @@
+class BinarySearchTree:
+    """Binary search tree as a collection of nodes organized in an order manner. Each
+    node has a value greater than all of its left child nodes and less than all of its
+    right child nodes. Each node has three different fields, '*value*' which acts as
+    the key to be provided, '*left*' and '*right*' denominating both children of the
+    node.
+
+    :param int value: value of the node.
+    """
+
+    def __init__(self, value=None):
+        self.value = value
+        self.left = None
+        self.right = None
+
+    def insert(self, value):
+        """Add node to tree.
+
+        :param int value: value of the node to insert in the tree.
+        """
+        if self.value is None:
+            self.value = value
```

Comment:
I think if you do `self.value = BinarySearchTree(value)` here, the last 2 cases can remove the `else` and just call `.insert(value)` on the appropriate node.